src/stream.rs

```diff
@@ -1,101 +1,205 @@
 use std::io::{Read, Seek};
-use std::sync::{Arc, Weak};
+use std::marker::Sync;
+use std::sync::Arc;
 use std::{error, fmt};
 
 use crate::decoder;
-use crate::dynamic_mixer::{self, DynamicMixerController};
+use crate::dynamic_mixer::{mixer, Mixer, MixerSource};
 use crate::sink::Sink;
-use crate::source::Source;
 use cpal::traits::{DeviceTrait, HostTrait, StreamTrait};
-use cpal::{Sample, SupportedStreamConfig};
+use cpal::{
+    BufferSize, ChannelCount, FrameCount, Sample, SampleFormat, SampleRate, StreamConfig,
+    SupportedBufferSize,
+};
 
-/// `cpal::Stream` container. Also see the more useful `OutputStreamHandle`.
+const HZ_44100: cpal::SampleRate = cpal::SampleRate(44_100);
+
+/// `cpal::Stream` container. Use `mixer()` method to control output.
 ///
-/// If this is dropped playback will end & attached `OutputStreamHandle`s will no longer work.
+/// If this is dropped, playback will end, and the associated output stream will be disposed.
 pub struct OutputStream {
-    mixer: Arc<DynamicMixerController<f32>>,
+    mixer: Arc<Mixer<f32>>,
     _stream: cpal::Stream,
 }
 
-/// More flexible handle to a `OutputStream` that provides playback.
-#[derive(Clone)]
-pub struct OutputStreamHandle {
-    mixer: Weak<DynamicMixerController<f32>>,
+impl OutputStream {
+    pub fn mixer(&self) -> Arc<Mixer<f32>> {
```

Comment:
I think it is the most straightforward choice to let users know that a mixer is the output. There is no need to explain that all sounds and sources added to output stream will be played concurrently. Also since the mixer is always there let it be actually useful.

Reply:
:+1: